source/numerics/matrix_creator.inst.in

```diff
@@ -305,80 +305,42 @@ for (scalar : COMPLEX_SCALARS; deal_II_dimension : DIMENSIONS; deal_II_space_dim
+      void MatrixCreator::create_laplace_matrix<deal_II_dimension,deal_II_space_dimension>
+      (const Mapping<deal_II_dimension,deal_II_space_dimension>       &mapping,
+       const DoFHandler<deal_II_dimension,deal_II_space_dimension>    &dof,
        const Quadrature<deal_II_dimension>    &q,
        SparseMatrix<double>     &matrix,
-       const Function<deal_II_dimension> * const coefficient,
+       const Function<deal_II_space_dimension,double> * const coefficient,
        const ConstraintMatrix   &constraints);
     template
-      void MatrixCreator::create_laplace_matrix<deal_II_dimension>
-      (const Mapping<deal_II_dimension>       &mapping,
-       const DoFHandler<deal_II_dimension>    &dof,
+      void MatrixCreator::create_laplace_matrix<deal_II_dimension,deal_II_space_dimension>
+      (const DoFHandler<deal_II_dimension,deal_II_space_dimension>    &dof,
        const Quadrature<deal_II_dimension>    &q,
        SparseMatrix<double>     &matrix,
-       const Function<deal_II_dimension> * const coefficient,
+       const Function<deal_II_space_dimension,double> * const coefficient,
        const ConstraintMatrix   &constraints);
     template
-      void MatrixCreator::create_laplace_matrix<deal_II_dimension>
-      (const Mapping<deal_II_dimension>       &mapping,
-       const DoFHandler<deal_II_dimension>    &dof,
+      void MatrixCreator::create_laplace_matrix<deal_II_dimension,deal_II_space_dimension>
+      (const Mapping<deal_II_dimension,deal_II_space_dimension>       &mapping,
+       const DoFHandler<deal_II_dimension,deal_II_space_dimension>    &dof,
        const Quadrature<deal_II_dimension>    &q,
        SparseMatrix<double>     &matrix,
-       const Function<deal_II_dimension>      &rhs,
+       const Function<deal_II_space_dimension,double>      &rhs,
        Vector<double>           &rhs_vector,
-       const Function<deal_II_dimension> * const coefficient,
+       const Function<deal_II_space_dimension,double> * const coefficient,
        const ConstraintMatrix   &constraints);
     template
-      void MatrixCreator::create_laplace_matrix<deal_II_dimension>
-      (const DoFHandler<deal_II_dimension>    &dof,
+      void MatrixCreator::create_laplace_matrix<deal_II_dimension,deal_II_space_dimension>
+      (const DoFHandler<deal_II_dimension,deal_II_space_dimension>    &dof,
        const Quadrature<deal_II_dimension>    &q,
        SparseMatrix<double>     &matrix,
-       const Function<deal_II_dimension>      &rhs,
+       const Function<deal_II_space_dimension,double>      &rhs,
        Vector<double>           &rhs_vector,
-       const Function<deal_II_dimension> * const coefficient,
+       const Function<deal_II_space_dimension,double> * const coefficient,
        const ConstraintMatrix   &constraints);
-
-// hp versions of create_laplace_matrix
-    template
-      void MatrixCreator::create_laplace_matrix<deal_II_dimension>
-      (const hp::DoFHandler<deal_II_dimension>    &dof,
```

Comment:
Yep clang is telling me that:
'create_laplace_matrix<1, 1>'

counts as:
create_laplace_matrix< 1 >

today i'll check if gcc agrees...